diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 33faee7..18345d1 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -70,7 +70,7 @@ add_library(convenience_static STATIC
 )
 target_include_directories(convenience_static
   PRIVATE ${CMAKE_SOURCE_DIR}/include)
-if(WIN32)
+if(WIN32 OR MINGW)
 add_library(libgetopt_static STATIC
     getopt/getopt.c
 )
@@ -96,10 +96,12 @@ target_link_libraries(rtl_sdr rtlsdr convenience_static
     ${LIBUSB_LIBRARIES}
     ${CMAKE_THREAD_LIBS_INIT}
 )
+
 target_link_libraries(rtl_tcp rtlsdr convenience_static
     ${LIBUSB_LIBRARIES}
     ${CMAKE_THREAD_LIBS_INIT}
 )
+
 target_link_libraries(rtl_test rtlsdr convenience_static
     ${LIBUSB_LIBRARIES}
     ${CMAKE_THREAD_LIBS_INIT}
@@ -131,11 +133,15 @@ target_link_libraries(rtl_power m)
 if(APPLE OR CMAKE_SYSTEM MATCHES "OpenBSD")
     target_link_libraries(rtl_test m)
 else()
-    target_link_libraries(rtl_test m rt)
+	if(MINGW)
+		target_link_libraries(rtl_test m)
+	else()	
+		target_link_libraries(rtl_test m rt)
+	endif()
 endif()
 endif()
 
-if(WIN32)
+if(WIN32 OR MINGW)
 target_link_libraries(rtl_sdr libgetopt_static)
 target_link_libraries(rtl_tcp ws2_32 libgetopt_static)
 target_link_libraries(rtl_test libgetopt_static)
diff --git a/src/rtl_tcp.c b/src/rtl_tcp.c
index 562198f..d403217 100644
--- a/src/rtl_tcp.c
+++ b/src/rtl_tcp.c
@@ -38,6 +38,10 @@
 #include "getopt/getopt.h"
 #endif
 
+#ifdef __MINGW64__
+#include <ws2tcpip.h>
+#endif
+
 #include <pthread.h>
 
 #include "rtl-sdr.h"
